src/Language/Haskell/TH/Lift/Generics.hs

```diff
@@ -251,27 +250,32 @@ instance (GLiftDatatype f, GLiftDatatype g) => GLiftDatatype (f :+: g) where
     gliftWith pName mName (L1 l) = gliftWith pName mName l
     gliftWith pName mName (R1 r) = gliftWith pName mName r
 
--- | Class of generic representation types which can be converted to a list of
--- Template Haskell expressions (which represent a constructors' arguments). You
--- shouldn't need to use this typeclass directly; it is only exported for educational
--- purposes.
+-- | Class of generic representation types which can conceptually be converted
+-- to a list of Template Haskell expressions (which represent a constructors'
+-- arguments). You shouldn't need to use this typeclass directly; it is only
+-- exported for educational purposes.
 class GLiftArgs f where
-    gliftArgs :: Quote m => f a -> [m Exp]
+    -- | @gliftArgs e f@ applies @f@ to the zero or more arguments represented
+    -- by @e@.
+    gliftArgs :: Quote m => f a -> Exp -> m Exp
 
 instance GLiftArgs U1 where
-    gliftArgs U1 = []
+    -- This pattern match must be strict, because
+    -- lift undefined really shouldn't just happen
+    -- to work for unit types.
+    gliftArgs U1 = return
```

Comment:
Should we use something like

```haskell
gliftArgs u = \x -> return (u `seq` x)
```

to match the behavior of the `V1` instance? Or (more to my taste) should we make the `V1` instance stricter to match `U1`? I don't see any *obvious* benefit to delaying the match.

Reply:
I'm not quite sure I understand the question, as there's no `GLiftArgs` instance for `V1`. Moreover, I'm unclear on what the difference between `gliftArgs U1 = return` and `gliftArgs u = \x -> return (u `seq` x)` is. I think I'd need to see concrete examples that demonstrate how they differ.